primitives/io/src/lib.rs

```diff
@@ -157,159 +183,219 @@ pub trait Storage {
-		self.kill_child_storage(storage_key, child_info);
-	}
-
-	/// Check whether the given `key` exists in storage.
-	fn exists(&self, key: &[u8]) -> bool {
-		self.exists_storage(key)
+		self.kill_child_storage(&child_info);
 	}
 
-	/// Check whether the given `key` exists in storage.
-	///
-	/// See `child_get` for common child api parameters.
+	/// Deprecated, please use dedicated runtime apis.
 	fn child_exists(
 		&self,
-		child_storage_key: &[u8],
+		storage_key: &[u8],
 		child_definition: &[u8],
 		child_type: u32,
 		key: &[u8],
 	) -> bool {
-		let storage_key = child_storage_key_or_panic(child_storage_key);
+		deprecated_storage_key_prefix_check(storage_key);
 		let child_info = ChildInfo::resolve_child_info(child_type, child_definition)
 			.expect("Invalid child definition");
-		self.exists_child_storage(storage_key, child_info, key)
+		self.exists_child_storage(&child_info, key)
 	}
 
-	/// Clear the storage of each key-value pair where the key starts with the given `prefix`.
-	fn clear_prefix(&mut self, prefix: &[u8]) {
-		Externalities::clear_prefix(*self, prefix)
-	}
-
-	/// Clear the child storage of each key-value pair where the key starts with the given `prefix`.
-	///
-	/// See `child_get` for common child api parameters.
+	/// Deprecated, please use dedicated runtime apis.
 	fn child_clear_prefix(
 		&mut self,
-		child_storage_key: &[u8],
+		storage_key: &[u8],
 		child_definition: &[u8],
 		child_type: u32,
 		prefix: &[u8],
 	) {
-		let storage_key = child_storage_key_or_panic(child_storage_key);
+		deprecated_storage_key_prefix_check(storage_key);
 		let child_info = ChildInfo::resolve_child_info(child_type, child_definition)
 			.expect("Invalid child definition");
-		self.clear_child_prefix(storage_key, child_info, prefix);
+		self.clear_child_prefix(&child_info, prefix);
 	}
 
-	/// "Commit" all existing operations and compute the resulting storage root.
-	///
-	/// The hashing algorithm is defined by the `Block`.
+	/// Deprecated, please use dedicated runtime apis.
+	fn child_root(
+		&mut self,
+		storage_key: &[u8],
+	) -> Vec<u8> {
+		let prefix = ChildType::ParentKeyId.parent_prefix();
+		if !storage_key.starts_with(prefix) {
+			panic!("Invalid storage key");
+		}
+		let storage_key = &storage_key[prefix.len()..];
+		let child_info = ChildInfo::resolve_child_info(ChildType::ParentKeyId as u32, storage_key)
+			.expect("Invalid storage key");
+		self.child_storage_root(&child_info)
+	}
+
+	/// Deprecated, please use dedicated runtime apis.
+	fn child_next_key(
+		&mut self,
+		storage_key: &[u8],
+		child_definition: &[u8],
+		child_type: u32,
+		key: &[u8],
+	) -> Option<Vec<u8>> {
+		deprecated_storage_key_prefix_check(storage_key);
+		let child_info = ChildInfo::resolve_child_info(child_type, child_definition)
+			.expect("Invalid child definition");
+		self.next_child_storage_key(&child_info, key)
+	}
+
+}
+
+
+/// Interface for accessing the child storage for default child trie,
+/// from within the runtime.
```

Comment:
```suggestion
/// Child storage interface.
/// 
/// Interface for accessing the child storage for default child trie,
/// from within the runtime.
```

And in general, good doc comments format which then renders nicely in html is

```rust
/// Header
/// 
/// Detailed description
```

(plenty of this requires some love in `sp-io`)

Reply:
That's good advice, thanks.